atomic_reactor/plugins/pre_check_user_settings.py

```diff
@@ -52,10 +57,35 @@ def dockerfile_checks(self):
                 "without user Dockerfile")
             return
 
+        self.label_version_check()
         self.appregistry_bundle_label_mutually_exclusive()
         self.operator_bundle_from_scratch()
         self.multistage_docker_api_check()
 
+    def label_version_check(self):
+        """Check that Dockerfile version has correct name. It means its value
+           can only contain alphanumeric characters, underscores, and dots between keywords.
+
+           Skip the check if there is no defined version
+        """
+        msg = (
+            "Dockerfile label version can only contain "
+            "alphanumeric characters, underscores, and dots between keywords"
+        )
+
+        parser = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
```

Comment:
you should also take in account flatpak, which doesn't have Dockerfile

Reply:
It's already prevented [here](https://github.com/containerbuildsystem/atomic-reactor/pull/1637/files#diff-f03b0cdbe34622ae6254586f69ab8d5aa2784be9e59b48e44bfb769dbd41152fL49)

Reply:
???

if in df_path isn't Dockerfile (flatpak), parser.labels will fail with : FileNotFoundError: [Errno 2] No such file or directory: './Dockerfile'

Reply:
@rcerven  self.label_version_check is called in function dockerfile_checks(), which returns at the start if self.flatpak is true

Reply:
oh ic